cli/src/main.rs

```diff
@@ -131,38 +139,86 @@ fn run() -> Result<(), CliError> {
                 (@arg key: -k --key +takes_value "Signing key name")
             )
         )
-    ).get_matches();
-
-    let (batch_link, mut wait) = if let Some(upload_matches) = matches.subcommand_matches("upload")
-    {
-        upload(upload_matches)?
-    } else if let Some(exec_matches) = matches.subcommand_matches("exec") {
-        execute(exec_matches)?
-    } else if let Some(ns_matches) = matches.subcommand_matches("ns") {
-        namespace_registry(ns_matches)?
-    } else if let Some(perm_matches) = matches.subcommand_matches("perm") {
-        namespace_permission(perm_matches)?
-    } else if let Some(cr_matches) = matches.subcommand_matches("cr") {
-        contract_registry(cr_matches)?
-    } else if let Some(sp_matches) = matches.subcommand_matches("sp") {
-        smart_permission(sp_matches)?
+    );
+
+    let app = app.subcommand(
```

Comment:
Are these meant to be behind a feature?  Otherwise, I'm not sure why they just weren't included in the `clap_app` macro call.

Reply:
List requires a set of possible values which cannot be provided using the clap_app, show was added the same way as list since it belongs to the same subcommand.